src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -769,15 +769,24 @@ def is_weight_compression_supported(
 
         return is_supported_dtype and not no_bit_reduction
 
-    def apply(
+    def get_weight_compression_parameters(
         self,
         model: TModel,
         graph: NNCFGraph,
         statistic_points: Optional[StatisticPointsContainer] = None,
         dataset: Optional[Dataset] = None,
-    ) -> TModel:
-        self.set_backend_entity(model)
+    ) -> tuple[list[WeightCompressionParameters], dict[str, Any]]:
+        """
+        Generates a list of weight compression parameters based on the Weight Compression algorithm
+        configuration. Determines the appropriate precision, group size, and other parameters for
+        each node eligible for weight compression.
 
+        :param model: Backend-specific input model.
+        :param graph: NNCFGraph instance.
+        :param statistic_points: Optional pre-collected statistic points.
+        :param dataset: Optional dataset for statistics collection.
+        :return: A tuple of weight compression parameters based on the Weight Compression algorithm configuration.
```

Comment:
:return: A tuple consisting of a list of weight compression parameters, based on the Weight Compression algorithm configuration, and a mapping of target node names to the collected statistics.